lib/ansible/module_utils/cloudfront.py

```diff
@@ -0,0 +1,122 @@
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author:
+#   - Willem van Ketwich <willem@vanketwich.com.au>
+#
+# Common functionality to be used by the modules:
+#   - cloudfront_distribution
+#   - cloudfront_invalidation
+#   - cloudfront_origin_access_identity
+
+
+"""
+Miscellaneous helpers for processing cloudfront data
+"""
+
+
+def change_dict_key_name(dictionary, old_key, new_key):
+    if old_key in dictionary:
+        dictionary[new_key] = dictionary.get(old_key)
+        dictionary.pop(old_key, None)
+    return dictionary
+
+
+def snake_dict_to_pascal_dict(snake_dict):
+    def pascalize(complex_type):
+        if complex_type is None:
+            return
+        new_type = type(complex_type)()
+        if isinstance(complex_type, dict):
+            for key in complex_type:
+                new_type[pascal(key)] = pascalize(complex_type[key])
+        elif isinstance(complex_type, list):
+            for i in range(len(complex_type)):
+                new_type.append(pascalize(complex_type[i]))
+        else:
+            return complex_type
+        return new_type
+
+    def pascal(words):
+        return (words.capitalize().split('_')[0] + ''.join(x.capitalize() or
+                '_' for x in words.split('_')[1:]))
+    return pascalize(snake_dict)
+
+
+def pascal_dict_to_snake_dict(pascal_dict, split_caps=False):
+    def pascal_to_snake(name):
+        import re
+        first_cap_re = re.compile('(.)([A-Z][a-z]+)')
+        all_cap_re = re.compile('([a-z0-9])([A-Z]+)')
+        split_cap_re = re.compile('([A-Z])')
+        s1 = first_cap_re.sub(r'\1\2', name)
+        if split_caps:
+            s2 = split_cap_re.sub(r'_\1', s1).lower()
+            s2 = s2[1:] if s2[0] == '_' else s2
+        else:
+            s2 = all_cap_re.sub(r'\1_\2', s1).lower()
+        return s2
+
+    def value_is_list(pascal_list):
+        checked_list = []
+        for item in pascal_list:
+            if isinstance(item, dict):
+                checked_list.append(pascal_dict_to_snake_dict(item, split_caps))
+            elif isinstance(item, list):
+                checked_list.append(value_is_list(item))
+            else:
+                checked_list.append(item)
+        return checked_list
+    snake_dict = {}
+    for k, v in pascal_dict.items():
+        if isinstance(v, dict):
+            snake_dict[pascal_to_snake(k)] = pascal_dict_to_snake_dict(
+                v, split_caps)
+        elif isinstance(v, list):
+            snake_dict[pascal_to_snake(k)] = value_is_list(v)
+        else:
+            snake_dict[pascal_to_snake(k)] = v
+    return snake_dict
+
+
+def merge_validation_into_config(config, validated_node, node_name):
+    if validated_node is not None:
+        if isinstance(validated_node, dict):
+            config_node = config.get(node_name)
+            if config_node is not None:
+                config_node_items = config_node.items()
```

Comment:
Since in the else this is defaulting to an empty list I think this line should be config_node_items = list(config_node.items()) to be python3 compatible and always get a list.